<!doctype html>
<html lang="zh"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta><meta name="theme-color" content="#123456"><meta name="generator" content="Hexo 4.2.0"><title>BELKA_2024 - Hello, NilEra :-)</title><link rel="manifest" href="/manifest.json"><meta name="application-name" content="Hello, NilEra :-)"><meta name="msapplication-TileImage" content="/img/StarLogo.svg"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-title" content="Hello, NilEra :-)"><meta name="apple-mobile-web-app-status-bar-style" content="default"><meta name="description" content="[TOC] Leash Bio - Predict New Medicines with BELKA用 BELKA 预测新药 Predict small molecule-protein interactions using the Big Encoded Library for Chemical Assessment (BELKA) 使用化学评估大编码库（BELKA）预测小分子蛋白质相互作用 O"><meta property="og:type" content="blog"><meta property="og:title" content="BELKA_2024"><meta property="og:url" content="https://hello-nilera.com/2024/05/19/BELKA-2024/"><meta property="og:site_name" content="Hello, NilEra :-)"><meta property="og:description" content="[TOC] Leash Bio - Predict New Medicines with BELKA用 BELKA 预测新药 Predict small molecule-protein interactions using the Big Encoded Library for Chemical Assessment (BELKA) 使用化学评估大编码库（BELKA）预测小分子蛋白质相互作用 O"><meta property="og:locale" content="zh_CN"><meta property="og:image" content="https://www.googleapis.com/download/storage/v1/b/kaggle-user-content/o/inbox%2F1095143%2F1901c6caa0c6c011617f4dec525d7bbe%2FKaggle%20v2%20(1).png?generation=1712179256934503&amp;alt=media"><meta property="article:published_time" content="2024-05-19T08:33:06.000Z"><meta property="article:modified_time" content="2024-07-29T07:32:43.979Z"><meta property="article:author" content="NilEra"><meta property="article:tag" content="深度学习"><meta property="article:tag" content="机器学习"><meta property="article:tag" content="PyTorch"><meta property="article:tag" content="Kaggle"><meta property="twitter:card" content="summary"><meta property="twitter:image:src" content="https://www.googleapis.com/download/storage/v1/b/kaggle-user-content/o/inbox%2F1095143%2F1901c6caa0c6c011617f4dec525d7bbe%2FKaggle%20v2%20(1).png?generation=1712179256934503&amp;alt=media"><script type="application/ld+json">{"@context":"https://schema.org","@type":"BlogPosting","mainEntityOfPage":{"@type":"WebPage","@id":"https://hello-nilera.com/2024/05/19/BELKA-2024/"},"headline":"BELKA_2024","image":[],"datePublished":"2024-05-19T08:33:06.000Z","dateModified":"2024-07-29T07:32:43.979Z","author":{"@type":"Person","name":"NilEra"},"publisher":{"@type":"Organization","name":"Hello, NilEra :-)","logo":{"@type":"ImageObject","url":"https://hello-nilera.com/img/StarLogo.svg"}},"description":"[TOC] Leash Bio - Predict New Medicines with BELKA用 BELKA 预测新药 Predict small molecule-protein interactions using the Big Encoded Library for Chemical Assessment (BELKA) 使用化学评估大编码库（BELKA）预测小分子蛋白质相互作用 O"}</script><link rel="canonical" href="https://hello-nilera.com/2024/05/19/BELKA-2024/"><link rel="icon" href="/img/StarLogo.svg"><link rel="stylesheet" href="https://cdnjs.loli.net/ajax/libs/font-awesome/6.0.0/css/all.min.css"><link rel="stylesheet" href="https://cdnjs.loli.net/ajax/libs/highlight.js/11.7.0/styles/atom-one-light.min.css"><link rel="stylesheet" href="https://fonts.loli.net/css2?family=Ubuntu:wght@400;600&amp;family=Source+Code+Pro"><link rel="stylesheet" href="/css/default.css"><style>body>.footer,body>.navbar,body>.section{opacity:0}</style><!--!--><script>var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "//hm.baidu.com/hm.js?249654dcf9a3bf70708fdfc6e2b1ec2b";
            var s = document.getElementsByTagName("script")[0];
            s.parentNode.insertBefore(hm, s);
        })();</script><meta name="msvalidate.01" content="F6BD78C6BD0096D2218CF88334111125"><!--!--><link rel="stylesheet" href="https://cdnjs.loli.net/ajax/libs/cookieconsent/3.1.1/cookieconsent.min.css"><link rel="stylesheet" href="https://cdnjs.loli.net/ajax/libs/lightgallery/1.10.0/css/lightgallery.min.css"><link rel="stylesheet" href="https://cdnjs.loli.net/ajax/libs/justifiedGallery/3.8.1/css/justifiedGallery.min.css"><!--!--><!--!--><!--!--><style>.pace{-webkit-pointer-events:none;pointer-events:none;-webkit-user-select:none;-moz-user-select:none;user-select:none}.pace-inactive{display:none}.pace .pace-progress{background:#3273dc;position:fixed;z-index:2000;top:0;right:100%;width:100%;height:2px}</style><script src="https://cdnjs.loli.net/ajax/libs/pace/1.2.4/pace.min.js"></script><!--!--><!--!--><!-- hexo injector head_end start --><script>
  (function () {
      function switchTab() {
          if (!location.hash) {
            return;
          }

          const id = '#' + CSS.escape(location.hash.substring(1));
          const $tabMenu = document.querySelector(`.tabs a[href="${id}"]`);
          if (!$tabMenu) {
            return;
          }

          const $tabMenuContainer = $tabMenu.parentElement.parentElement;
          Array.from($tabMenuContainer.children).forEach($menu => $menu.classList.remove('is-active'));
          Array.from($tabMenuContainer.querySelectorAll('a'))
              .map($menu => document.getElementById($menu.getAttribute("href").substring(1)))
              .forEach($content => $content.classList.add('is-hidden'));

          if ($tabMenu) {
              $tabMenu.parentElement.classList.add('is-active');
          }
          const $activeTab = document.querySelector(id);
          if ($activeTab) {
              $activeTab.classList.remove('is-hidden');
          }
      }
      switchTab();
      window.addEventListener('hashchange', switchTab, false);
  })();
  </script><!-- hexo injector head_end end --></head><body class="is-3-column"><nav class="navbar navbar-main"><div class="container navbar-container"><div class="navbar-brand justify-content-center"><a class="navbar-item navbar-logo" href="/"><img src="/img/StarLogo.svg" alt="Hello, NilEra :-)" height="28"></a></div><div class="navbar-menu"><div class="navbar-start"><a class="navbar-item" href="/">首页</a><a class="navbar-item" href="/archives">归档</a><a class="navbar-item" href="/categories">目录</a><a class="navbar-item" href="/tags">标签</a><a class="navbar-item" href="/about">关于</a><a class="navbar-item" href="/me">我</a></div><div class="navbar-end"><a class="navbar-item" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/NilEra-K"><i class="fab fa-github"></i></a><a class="navbar-item is-hidden-tablet catalogue" title="目录" href="javascript:;"><i class="fas fa-list-ul"></i></a><a class="navbar-item search" title="搜索" href="javascript:;"><i class="fas fa-search"></i></a></div></div></div></nav><section class="section"><div class="container"><div class="columns"><div class="column order-2 column-main is-8-tablet is-8-desktop is-6-widescreen"><div class="card"><article class="card-content article" role="article"><div class="article-meta is-size-7 is-uppercase level is-mobile"><div class="level-left"><span class="level-item"><time dateTime="2024-05-19T08:33:06.000Z" title="2024/5/19 16:33:06">2024-05-19</time>发表</span><span class="level-item"><time dateTime="2024-07-29T07:32:43.979Z" title="2024/7/29 15:32:43">2024-07-29</time>更新</span><span class="level-item"><a class="link-muted" href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/">深度学习</a><span> / </span><a class="link-muted" href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/Kaggle/">Kaggle</a></span><span class="level-item">38 分钟读完 (大约5697个字)</span></div></div><h1 class="title is-3 is-size-4-mobile">BELKA_2024</h1><div class="content"><p>[TOC]</p>
<h1 id="Leash-Bio-Predict-New-Medicines-with-BELKA"><a href="#Leash-Bio-Predict-New-Medicines-with-BELKA" class="headerlink" title="Leash Bio - Predict New Medicines with BELKA"></a>Leash Bio - Predict New Medicines with BELKA</h1><p>用 <em><strong>BELKA</strong></em> 预测新药</p>
<p>Predict small molecule-protein interactions using the Big Encoded Library for Chemical Assessment (BELKA)</p>
<p>使用化学评估大编码库（BELKA）预测小分子蛋白质相互作用</p>
<h2 id="Overview"><a href="#Overview" class="headerlink" title="Overview"></a>Overview</h2><p>In this competition, you’ll develop machine learning (ML) models to predict the binding affinity of small molecules to specific protein targets – a critical step in drug development for the pharmaceutical industry that would pave the way for more accurate drug discovery. You’ll help predict which drug-like small molecules (chemicals) will bind to three possible protein targets.</p>
<p>在这场比赛中，你将开发机器学习（ML）模型来预测小分子与特定<strong>蛋白质靶标（目标蛋白）</strong>的结合亲和力——这是制药行业药物开发的关键一步，将为更准确的药物发现铺平道路。你将帮助预测哪种药物样的小分子（化学物质）将与三种可能的蛋白质靶点结合。</p>
<h2 id="Description"><a href="#Description" class="headerlink" title="Description"></a>Description</h2><p>Small molecule drugs are chemicals that interact with cellular protein machinery and affect the functions of this machinery in some way. Often, drugs are meant to inhibit the activity of single protein targets, and those targets are thought to be involved in a disease process. A classic approach to identify such candidate molecules is to physically make them, one by one, and then expose them to the protein target of interest and test if the two interact. This can be a fairly laborious and time-intensive process.</p>
<p><em>小分子药物是与细胞蛋白质机制相互作用并以某种方式影响该机制功能的化学物质。通常，药物旨在抑制单个蛋白质靶标的活性，而这些靶标被认为与疾病过程有关。识别这类候选分子的一种经典方法是一个接一个地进行物理制造，然后将其暴露于感兴趣的蛋白质靶点，并测试两者是否相互作用。这可能是一个相当费力和耗时的过程。</em></p>
<p>The US Food and Drug Administration (FDA) has approved roughly <a target="_blank" rel="noopener" href="https://www.fda.gov/drugs/development-approval-process-drugs/new-drugs-fda-cders-new-molecular-entities-and-new-therapeutic-biological-products">2,000 novel molecular entities</a> in its <a target="_blank" rel="noopener" href="https://www.fda.gov/about-fda/histories-fda-regulated-products/summary-nda-approvals-receipts-1938-present">entire history</a>. However, the number of chemicals in druglike space has been <a target="_blank" rel="noopener" href="https://www.nature.com/articles/432823a">estimated to be 10^60</a>, a space far too big to physically search. There are likely effective treatments for human ailments hiding in that chemical space, and better methods to find such treatments are desirable to us all.</p>
<p><em>美国食品药品监督管理局（FDA）已经批准了大约<a target="_blank" rel="noopener" href="https://www.fda.gov/drugs/development-approval-process-drugs/new-drugs-fda-cders-new-molecular-entities-and-new-therapeutic-biological-products">2000种新型分子实体</a>在其<a target="_blank" rel="noopener" href="https://www.fda.gov/about-fda/histories-fda-regulated-products/summary-nda-approvals-receipts-1938-present">整个历史</a>. 然而，类药物领域的化学物质数量<a target="_blank" rel="noopener" href="https://www.nature.com/articles/432823a">估计为$10^60$</a>，这个空间太大了，无法进行物理搜索。在这个化学空间里，可能有有效的治疗人类疾病的方法，而找到更好的治疗方法对我们所有人来说都是可取的。</em></p>
<p>To evaluate potential search methods in small molecule chemistry, competition host Leash Biosciences physically tested some 133M small molecules for their ability to interact with one of three protein targets using DNA-encoded chemical library (DEL) technology. This dataset, the Big Encoded Library for Chemical Assessment (BELKA), provides an excellent opportunity to develop predictive models that may advance drug discovery.</p>
<p><em>为了评估小分子化学中潜在的搜索方法，比赛主办方Leash Biosciences使用DNA编码化学文库（DEL）技术对约133M个小分子进行了物理测试，以确定它们与三个蛋白质靶标之一相互作用的能力。该数据集，即化学评估大编码库（BELKA），为开发可能促进药物发现的预测模型提供了极好的机会。</em></p>
<p>Datasets of this size are rare and restricted to large pharmaceutical companies. The current best-curated public dataset of this kind is perhaps <a target="_blank" rel="noopener" href="https://www.bindingdb.org/rwd/bind/index.jsp">bindingdb</a>, which, at 2.8M binding measurements, is much smaller than BELKA.</p>
<p><em>这种规模的数据集非常罕见，仅限于大型制药公司。目前这类最好的公共数据集可能是<a target="_blank" rel="noopener" href="https://www.bindingdb.org/rwd/bind/index.jsp">bindingdb</a>，在2.8M的结合测量值下，比BELKA小得多。</em></p>
<p>This competition aims to revolutionize small molecule binding prediction by harnessing ML techniques. Recent advances in ML approaches suggest it might be possible to search chemical space by inference using well-trained computational models rather than running <a target="_blank" rel="noopener" href="https://blog.research.google/2020/06/unlocking-chemome-with-dna-encoded.html">laboratory</a> <a target="_blank" rel="noopener" href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.0c00452">experiments</a>. Similar progress in <a target="_blank" rel="noopener" href="https://proceedings.neurips.cc/paper_files/paper/2012/file/c399862d3b9d6b76c8436e924a68c45b-Paper.pdf">other</a> <a target="_blank" rel="noopener" href="https://arxiv.org/abs/1706.03762">fields</a> suggest using ML to search across vast spaces could be a generalizable approach applicable to many domains. We hope that by providing BELKA we will democratize aspects of computational drug discovery and assist the community in finding new lifesaving medicines.</p>
<p><em>这项竞赛旨在通过利用ML技术彻底改变小分子结合预测。ML方法的最新进展表明，使用训练有素的计算模型而不是进行<a target="_blank" rel="noopener" href="https://blog.research.google/2020/06/unlocking-chemome-with-dna-encoded.html">实验室</a> <a target="_blank" rel="noopener" href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.0c00452">实验</a>，通过推理搜索化学空间是可能的。<a target="_blank" rel="noopener" href="https://proceedings.neurips.cc/paper_files/paper/2012/file/c399862d3b9d6b76c8436e924a68c45b-Paper.pdf">其他</a> <a target="_blank" rel="noopener" href="https://arxiv.org/abs/1706.03762">领域</a>的类似进展表明，使用ML在广阔的空间中搜索可能是一种适用于许多领域的通用方法。我们希望通过提供BELKA，我们将使计算药物发现的各个方面民主化，并帮助社区寻找新的救命药物。</em></p>
<p>Here, you’ll build predictive models to estimate the binding affinity of unknown chemical compounds to specified protein targets. You may use the training data provided; alternatively, there are a number of methods to make small molecule binding predictions without relying on empirical binding data (e.g. <a target="_blank" rel="noopener" href="https://github.com/gcorso/DiffDock">DiffDock</a>, and this contest was designed to allow for such submissions).</p>
<p><em>在这里，你将建立预测模型来估计未知化合物与特定蛋白质靶标的结合亲和力。您可以使用提供的培训数据；或者，有许多方法可以在不依赖经验结合数据的情况下进行小分子结合预测（例如<a target="_blank" rel="noopener" href="https://github.com/gcorso/DiffDock">DiffDock</a>，而本次竞赛旨在允许此类提交）。</em></p>
<p>Your work will contribute to advances in small molecule chemistry used to accelerate drug discovery.</p>
<p><em>你的工作将有助于促进用于加速药物发现的小分子化学的进步。</em></p>
<h2 id="Evaluation"><a href="#Evaluation" class="headerlink" title="Evaluation"></a>Evaluation</h2><p>This metric for this competition is the average precision calculated for each (protein, split group) and then averaged for the final score. Please see this <a target="_blank" rel="noopener" href="https://www.kaggle.com/competitions/leash-BELKA/discussion/503232">forum post</a> for important details.</p>
<p><em>这项比赛的指标是为每个（蛋白质、分组）计算的平均精度，然后为最终得分取平均值。请参阅此<a target="_blank" rel="noopener" href="https://www.kaggle.com/competitions/leash-BELKA/discussion/503232">论坛帖子</a>了解重要细节。</em></p>
<p>Here’s the <a target="_blank" rel="noopener" href="https://www.kaggle.com/code/metric/leash-average-map">code</a> for the implementation.</p>
<p><em>这是<a target="_blank" rel="noopener" href="https://www.kaggle.com/code/metric/leash-average-map">代码</a>以供实施。</em></p>
<h3 id="Submission-File"><a href="#Submission-File" class="headerlink" title="Submission File"></a>Submission File</h3><p>For each <code>id</code> in the test set, you must predict a probability for the binary target <code>binds</code> target. The file should contain a header and have the following format:</p>
<p><em>对于测试集中的每个</em><code>id</code>，<em>您必须预测二进制目标“绑定”目标的概率。该文件应包含一个标头，并具有以下格式：</em></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">id,binds</span><br><span class="line">295246830,0.5</span><br><span class="line">295246831,0.5</span><br><span class="line">295246832,0.5</span><br><span class="line">etc.</span><br></pre></td></tr></table></figure>

<h2 id="Timeline"><a href="#Timeline" class="headerlink" title="Timeline"></a>Timeline</h2><ul>
<li><strong>April 4, 2024</strong> - Start Date.</li>
<li><strong>July 1, 2024</strong> - Entry Deadline. You must accept the competition rules before this date in order to compete.</li>
<li><strong>July 1, 2024</strong> - Team Merger Deadline. This is the last day participants may join or merge teams.</li>
<li><strong>July 8, 2024</strong> - Final Submission Deadline.</li>
</ul>
<p>All deadlines are at 11:59 PM UTC on the corresponding day unless otherwise noted. The competition organizers reserve the right to update the contest timeline if they deem it necessary.</p>
<h2 id="Prizes"><a href="#Prizes" class="headerlink" title="Prizes"></a>Prizes</h2><ul>
<li>First Prize: $12,000</li>
<li>Second Prize: $10,000</li>
<li>Third Prize: $10,000</li>
<li>Fourth Prize: $8,000</li>
<li>Fifth Prize: $5,000</li>
<li>Top Student Group: $5,000 to the highest performing student team. A team would be considered a student team if majority members (e.g. at least 3 out of a 5 member team) are students enrolled in a high school or university degree. In the case of an even number of members, half of them must be students.</li>
</ul>
<h2 id="Competition-Host"><a href="#Competition-Host" class="headerlink" title="Competition Host"></a>Competition Host</h2><p><a target="_blank" rel="noopener" href="https://www.leash.bio/">Leash Biosciences</a> is a discovery-stage biotechnology company that seeks to improve medicinal chemistry with machine learning approaches and massive data collection. Leash is comprised of wet lab scientists and dry lab scientists in equal numbers, and is proudly headquartered in Salt Lake City, Utah, USA.</p>
<h2 id="Additional-Details"><a href="#Additional-Details" class="headerlink" title="Additional Details"></a>Additional Details</h2><h3 id="Chemical-Representations"><a href="#Chemical-Representations" class="headerlink" title="Chemical Representations"></a>Chemical Representations</h3><p>One of the goals of this competition is to explore and compare many different ways of representing molecules. Small molecules <a target="_blank" rel="noopener" href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00460-5">have</a> <a target="_blank" rel="noopener" href="https://wires.onlinelibrary.wiley.com/doi/10.1002/wcms.1603">been</a> [represented](<a target="_blank" rel="noopener" href="https://pubs.acs.org/doi/10.1021/acsinfocus.7e7006?ref=infocus/AI_&">https://pubs.acs.org/doi/10.1021/acsinfocus.7e7006?ref=infocus%2FAI_&amp;</a> Machine Learning) <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10689004/">with</a> SMILES, graphs, 3D structures, and more, including more esoteric methods <a target="_blank" rel="noopener" href="https://arxiv.org/abs/1801.10130">such as spherical convolutional neural nets</a>. We encourage competitors to explore not only different methods of making predictions but also to try different ways of representing the molecules.</p>
<p>We provide the molecules in SMILES format.</p>
<p><em>这场比赛的目标之一是探索和比较许多不同的分子表现方式。小分子已经用SMILES、图形、3D结构等表示，包括更深奥的方法，如球形卷积神经网络。我们鼓励竞争对手不仅探索不同的预测方法，还尝试不同的分子表示方法。</em></p>
<p><em>我们提供SMILES格式的分子。</em></p>
<h3 id="SMILES"><a href="#SMILES" class="headerlink" title="SMILES"></a>SMILES</h3><p>SMILES is a concise string notation used to represent the structure of chemical molecules. It encodes the molecular graph, including atoms, bonds, connectivity, and stereochemistry as a linear sequence of characters, by traversing the molecule graph. SMILES is widely used in machine learning applications for chemistry, such as molecular property prediction, drug discovery, and materials design, as it provides a standardized and machine-readable format for representing and manipulating chemical structures.</p>
<p>The SMILES in this dataset should be sufficient to be translated into any other chemical representation format that you want to try. A simple way to perform some of these translations is with <a target="_blank" rel="noopener" href="https://www.rdkit.org/docs/GettingStartedInPython.html">RDKit</a>.</p>
<p><em>SMILES是一种简明的字符串表示法，用于表示化学分子的结构。它通过遍历分子图，将分子图（包括原子、键、连接性和立体化学）编码为线性字符序列。SMILES广泛用于化学的机器学习应用，如分子性质预测、药物发现和材料设计，因为它为表示和操纵化学结构提供了标准化和机器可读的格式。</em><br><em>该数据集中的SMILES应该足以转换为您想要尝试的任何其他化学表示格式。执行其中一些翻译的一种简单方法是使用<a target="_blank" rel="noopener" href="https://www.rdkit.org/docs/GettingStartedInPython.html">RDKit</a>.</em></p>
<h3 id="Details-about-the-experiments"><a href="#Details-about-the-experiments" class="headerlink" title="Details about the experiments"></a>Details about the experiments</h3><h3 id="DELs-are-libraries-of-small-molecules-with-unique-DNA-barcodes-covalently-attached"><a href="#DELs-are-libraries-of-small-molecules-with-unique-DNA-barcodes-covalently-attached" class="headerlink" title="DELs are libraries of small molecules with unique DNA barcodes covalently attached"></a>DELs are libraries of small molecules with unique DNA barcodes covalently attached</h3><p>Traditional <a target="_blank" rel="noopener" href="https://www.nature.com/articles/nrd3368">high-throughput screening</a> requires keeping individual small molecules in separate, identifiable tubes and demands a lot of liquid handling to test each one of those against the protein target of interest in a separate reaction. The logistical overhead of these efforts tends to restrict screening collections, called libraries, to 50K-5M small molecules. A scalable solution to this problem, DNA-encoded chemical libraries, was <a target="_blank" rel="noopener" href="https://www.nature.com/articles/nchembio.211">described in 2009</a>. As DNA sequencing got <a target="_blank" rel="noopener" href="https://www.genome.gov/about-genomics/fact-sheets/Sequencing-Human-Genome-cost">cheaper and cheaper</a>, it became clear that DNA itself could be used as a label to identify, and deconvolute, collections of molecules in a complex mixture. <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41573-023-00713-6">DELs</a> <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8369695/">leverage</a> this DNA sequencing technology.</p>
<p>These barcoded small molecules are in a pool (many in a single tube, rather than one tube per small molecule) and are exposed to the protein target of interest in solution. The protein target of interest is then rinsed to remove small molecules in the DEL that don’t bind the target, and the remaining binders are collected and their DNA sequenced.</p>
<p><strong>DEL是共价连接有独特DNA条形码的小分子库</strong><br><em>传统<a target="_blank" rel="noopener" href="https://www.nature.com/articles/nrd3368">高通量筛选</a>需要将单个小分子保持在单独的、可识别的管中，并且需要大量的液体处理来在单独的反应中针对感兴趣的蛋白质靶标测试其中的每一个。这些工作的后勤开销往往将筛选收藏（称为文库）限制在5000万至500万个小分子以内。这个问题的一个可扩展的解决方案，DNA编码的化学文库，<a target="_blank" rel="noopener" href="https://www.nature.com/articles/nchembio.211">在2009年描述</a>. 随着DNA测序变得<a target="_blank" rel="noopener" href="https://www.genome.gov/about-genomics/fact-sheets/Sequencing-Human-Genome-cost">越来越便宜</a>，很明显，DNA本身可以用作标签来识别和消除复杂混合物中分子的聚集。<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41573-023-00713-6%E5%BD%B1%E5%93%8D%E5%8A%9Bhttps://www.ncbi.nlm.nih.gov/pmc/articles/PMC8369695/">DELs</a>这种DNA测序技术。</em><br><em>这些条形码小分子在一个池中（许多在单管中，而不是每个小分子一管），并暴露于溶液中感兴趣的蛋白质靶标。然后冲洗感兴趣的蛋白质靶标，以去除DEL中不与靶标结合的小分子，收集剩余的结合物并对其DNA进行测序。</em></p>
<h3 id="DELs-are-manufactured-by-combining-different-building-blocks"><a href="#DELs-are-manufactured-by-combining-different-building-blocks" class="headerlink" title="DELs are manufactured by combining different building blocks"></a>DELs are manufactured by combining different building blocks</h3><p>An intuitive way to think about DELs is to imagine a Mickey Mouse head as an example of a small molecule in the DEL. We attach the DNA barcode to Mickey’s chin. Mickey’s left ear is connected by a zipper; Mickey’s right ear is connected by velcro. These attachment points of zippers and velcro are analogies to different chemical reactions one might use to construct the DEL.</p>
<p>We could purchase ten different Mickey Mouse faces, ten different zipper ears, and ten different velcro ears, and use them to construct our small molecule library. By creating every combination of these three, we’ll have 1,000 small molecules, but we only needed thirty building blocks (faces and ears) to make them. This combinatorial approach is what allows DELs to have so many members: the library in this competition is composed of 133M small molecules. The 133M small molecule library used here, AMA014, was provided by <a target="_blank" rel="noopener" href="https://www.alphama.com.cn/en/about/">AlphaMa</a>. It has a triazine core and superficially resembles the DELs described <a target="_blank" rel="noopener" href="https://www.nature.com/articles/nchembio.211">here</a>.</p>
<p><strong>DEL是通过组合不同的构建块来制造的</strong><br>一个思考DEL的直观方法是想象一个米老鼠的头作为DEL中一个小分子的例子。我们把DNA条形码贴在米奇的下巴上。米奇的左耳由拉链连接；米奇的右耳是用尼龙搭扣连接的。拉链和尼龙搭扣的这些连接点类似于可能用于构建DEL的不同化学反应。<br>我们可以购买十个不同的米老鼠脸、十个不同拉链耳朵和十个不同尼龙搭扣耳朵，并用它们来构建我们的小分子库。通过创建这三者的每一个组合，我们将拥有1000个小分子，但我们只需要30个构建块（脸和耳朵）就可以制造它们。这种组合方法使DEL能够拥有如此多的成员：这场竞争中的文库由133M个小分子组成。这里使用的133M小分子文库AMA014由<a target="_blank" rel="noopener" href="https://www.alphama.com.cn/en/about/">AlphaMa</a>提供。它有一个三嗪核心，表面上类似于<a target="_blank" rel="noopener" href="https://www.nature.com/articles/nchembio.211">此处</a>描述的DEL。</p>
<img src="https://www.googleapis.com/download/storage/v1/b/kaggle-user-content/o/inbox%2F1095143%2F1901c6caa0c6c011617f4dec525d7bbe%2FKaggle%20v2%20(1).png?generation=1712179256934503&alt=media">

<h2 id="Acknowledgements"><a href="#Acknowledgements" class="headerlink" title="Acknowledgements"></a>Acknowledgements</h2><p>Leash Biosciences is grateful for the generous cosponsorship of <a target="_blank" rel="noopener" href="https://www.topharvestcap.com/">Top Harvest Capital</a> and <a target="_blank" rel="noopener" href="https://www.alphama.com.cn/en/about/">AlphaMa</a>.</p>
<h2 id="Citation"><a href="#Citation" class="headerlink" title="Citation"></a>Citation</h2><p>Andrew Blevins, Ian K Quigley, Brayden J Halverson, Nate Wilkinson, Rebecca S Levin, Agastya Pulapaka, Walter Reade, Addison Howard. (2024). Leash Bio - Predict New Medicines with BELKA. Kaggle. <a target="_blank" rel="noopener" href="https://kaggle.com/competitions/leash-BELKA">https://kaggle.com/competitions/leash-BELKA</a></p>
<hr>
<h2 id="Dataset-Description"><a href="#Dataset-Description" class="headerlink" title="Dataset Description"></a>Dataset Description</h2><h2 id="Overview-1"><a href="#Overview-1" class="headerlink" title="Overview"></a><strong>Overview</strong></h2><p>The examples in the competition dataset are represented by a binary classification of whether a given small molecule is a binder or not to one of three protein targets. The data were collected using DNA-encoded chemical library (DEL) technology.</p>
<p><em>比赛数据集中的例子由给定小分子是否与三个蛋白质靶标之一结合的二元分类表示。使用DNA编码化学文库（DEL）技术收集数据。</em></p>
<p>We represent chemistry with SMILES (<a target="_blank" rel="noopener" href="https://archive.epa.gov/med/med_archive_03/web/html/smiles.html">Simplified Molecular-Input Line-Entry System</a>) and the labels as binary binding classifications, one per protein target of three targets.</p>
<p>我们用SMILES（<a target="_blank" rel="noopener" href="https://archive.epa.gov/med/med_archive_03/web/html/smiles.html">简化分子输入 行输入系统</a>)和二元绑定分类来表示化学，三个靶标中的每个蛋白质靶标都有一个。</p>
<h2 id="Files"><a href="#Files" class="headerlink" title="Files"></a><strong>Files</strong></h2><p><strong>[train&#x2F;test].[csv&#x2F;parquet]</strong> - The train or test data, available in both the csv and parquet formats.</p>
<ul>
<li><code>id</code> - A unique example_id that we use to identify the molecule-binding target pair.</li>
<li><code>buildingblock1_smiles</code> - The structure, in SMILES, of the first building block</li>
<li><code>buildingblock2_smiles</code> - The structure, in SMILES, of the second building block</li>
<li><code>buildingblock3_smiles</code> - The structure, in SMILES, of the third building block</li>
<li><code>molecule_smiles</code> - The structure of the fully assembled molecule, in SMILES. This includes the three building blocks and the triazine core. Note we use a <code>[Dy]</code> as the stand-in for the DNA linker.</li>
<li><code>protein_name</code> - The protein target name</li>
<li><code>binds</code> - The target column. A binary class label of whether the molecule binds to the protein. Not available for the test set.</li>
</ul>
<p><strong>sample_submission.csv</strong> - A sample submission file in the correct format</p>
<p><strong>[train&#x2F;test].[csv&#x2F;parquet]</strong> - 训练或测试数据，csv和parquet格式均可。</p>
<ul>
<li><code>id</code> - 我们用来识别分子结合靶标对的唯一示例_id。</li>
<li><code>buildingblock1_smiles</code> - 第一个构建块的结构，以SMILES表示</li>
<li><code>buildingblock2_smiles</code> - 第二个构建块的结构，以SMILES表示</li>
<li><code>buildingblock3_smiles</code> - 第三个构建块的结构，以SMILES表示</li>
<li><code>molecule_smiles</code> - 完全组装的分子的结构，以SMILES表示。这包括三个构建块和三嗪核心。请注意，我们使用<code>[Dy]</code>作为DNA连接子的替代。</li>
<li><code>protein_name</code> - 蛋白质靶标名称</li>
<li><code>binds</code> - 目标列。分子是否与蛋白质结合的二进制类标签。不适用于测试集。</li>
</ul>
<h2 id="Competition-data"><a href="#Competition-data" class="headerlink" title="Competition data"></a><strong>Competition data</strong></h2><p>All data were generated in-house at Leash Biosciences. We are providing roughly 98M training examples per protein, 200K validation examples per protein, and 360K test molecules per protein. To test generalizability, the test set contains building blocks that are not in the training set. These datasets are very imbalanced: roughly 0.5% of examples are classified as binders; we used 3 rounds of selection in triplicate to identify binders experimentally. Following the competition, Leash will make all the data available for future use (3 targets × 3 rounds of selection × 3 replicates × 133M molecules, or 3.6B measurements).</p>
<p><em>所有数据均由Leash Biosciences公司内部生成。我们为每种蛋白质提供了大约 98M 个训练实例，为每种蛋白提供了 200K 个验证实例，为每个蛋白质提供了 360K 个测试分子。为了测试可推广性，测试集包含不在训练集中的构建块。这些数据集非常不平衡：大约0.5%的示例被归类为绑定；我们使用了三轮一式三份的选择来实验鉴定粘合剂。比赛结束后，Leash将提供所有数据供未来使用（3个靶标×3轮选择×3个重复×3.33M个分子，或3.6B测量值）。</em></p>
<h2 id="Targets"><a href="#Targets" class="headerlink" title="Targets"></a><strong>Targets</strong></h2><p>Proteins are encoded in the genome, and names of the genes encoding those proteins are typically bestowed by their discoverers and regulated by the <a target="_blank" rel="noopener" href="https://www.genenames.org/">Hugo Gene Nomenclature Committee</a>. The protein products of these genes can sometimes have different names, often due to the history of their discovery.</p>
<p>We screened three protein targets for this competition.</p>
<p>蛋白质在基因组中编码，编码这些蛋白质的基因的名称通常由其发现者命名，并由<a target="_blank" rel="noopener" href="https://www.genenames.org/">雨果基因命名委员会</a>监管。这些基因的蛋白质产物有时可能有不同的名称，通常是由于它们的发现历史。<br>我们为这次比赛筛选了三个蛋白质靶点。</p>
<h3 id="EPHX2-sEH"><a href="#EPHX2-sEH" class="headerlink" title="EPHX2 (sEH)"></a><strong>EPHX2 (sEH)</strong></h3><p>The first target, epoxide hydrolase 2, is encoded by the EPHX2 genetic locus, and its protein product is commonly named “soluble epoxide hydrolase”, or abbreviated to sEH. Hydrolases are enzymes that catalyze certain chemical reactions, and EPHX2&#x2F;sEH also hydrolyzes certain phosphate groups. EPHX2&#x2F;sEH is a potential drug target for high blood pressure and diabetes progression, and small molecules inhibiting EPHX2&#x2F;sEH from earlier DEL efforts <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8429121/">made it to clinical trials</a>.</p>
<p>EPHX2&#x2F;sEH was also screened with DELs, and hits predicted with ML approaches, in a <a target="_blank" rel="noopener" href="https://blog.research.google/2020/06/unlocking-chemome-with-dna-encoded.html">recent</a> <a target="_blank" rel="noopener" href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.0c00452">study</a> but the screening data were not published. We included EPHX2&#x2F;sEH to allow contestants an external gut check for model performance by comparing to these previously-published results.</p>
<p>We screened EPHX2&#x2F;sEH <a target="_blank" rel="noopener" href="https://www.caymanchem.com/product/10011669/soluble-epoxide-hydrolase-(human-recombinant)">purchased from Cayman Chemical</a>, a life sciences commercial vendor. For those contestants wishing to incorporate protein structural information in their submissions, the amino sequence is positions 2-555 from UniProt entry <a target="_blank" rel="noopener" href="https://www.uniprot.org/uniprotkb/P34913/entry#sequences">P34913</a>, the crystal structure can be found in <a target="_blank" rel="noopener" href="https://www.rcsb.org/structure/3i28">PDB entry 3i28</a>, and predicted structure can be found in AlphaFold2 entry <a target="_blank" rel="noopener" href="https://alphafold.ebi.ac.uk/entry/P34913">34913</a>. Additional EPHX2&#x2F;sEH crystal structures with ligands bound can be found in PDB.</p>
<p>第一个靶标环氧化物水解酶2由EPHX2基因座编码，其蛋白产物通常被命名为“可溶性环氧化物水解酶”，或缩写为sEH。水解酶是催化某些化学反应的酶，EPHX2&#x2F;sEH也水解某些磷酸基团。EPHX2&#x2F;sEH是高血压和糖尿病进展的潜在药物靶点，早期DEL研究中抑制EPHX2&#x2F;s EH的小分子<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8429121/">已进入临床试验</a>.<br>EPHX2&#x2F;sEH也用DEL进行了筛选，并用ML方法预测了命中率(<a target="_blank" rel="noopener" href="https://blog.research.google/2020/06/unlocking-chemome-with-dna-encoded.html%E5%AD%A6%E4%B9%A0https://pubs.acs.org/doi/10.1021/acs.jmedchem.0c00452)%E4%BD%86%E7%AD%9B%E9%80%89%E6%95%B0%E6%8D%AE%E6%B2%A1%E6%9C%89%E5%85%AC%E5%B8%83%E3%80%82%E6%88%91%E4%BB%AC%E7%BA%B3%E5%85%A5%E4%BA%86EPHX2/sEH%EF%BC%8C%E9%80%9A%E8%BF%87%E4%B8%8E%E4%B9%8B%E5%89%8D%E5%85%AC%E5%B8%83%E7%9A%84%E7%BB%93%E6%9E%9C%E8%BF%9B%E8%A1%8C%E6%AF%94%E8%BE%83%EF%BC%8C%E8%AE%A9%E5%8F%82%E8%B5%9B%E8%80%85%E8%83%BD%E5%A4%9F%E5%AF%B9%E6%A8%A1%E5%9E%8B%E6%80%A7%E8%83%BD%E8%BF%9B%E8%A1%8C%E5%A4%96%E9%83%A8%E6%A3%80%E6%9F%A5%E3%80%82">https://blog.research.google/2020/06/unlocking-chemome-with-dna-encoded.html学习https://pubs.acs.org/doi/10.1021/acs.jmedchem.0c00452)但筛选数据没有公布。我们纳入了EPHX2/sEH，通过与之前公布的结果进行比较，让参赛者能够对模型性能进行外部检查。</a><br>我们筛选了EPHX2&#x2F;sEH<a target="_blank" rel="noopener" href="https://www.caymanchem.com/product/10011669/soluble-epoxide-hydrolase-%EF%BC%88%E4%BA%BA%E7%B1%BB%E9%87%8D%E7%BB%84%EF%BC%89%EF%BC%89%EF%BC%8C%E4%B8%80%E5%AE%B6%E7%94%9F%E5%91%BD%E7%A7%91%E5%AD%A6%E5%95%86%E4%B8%9A%E4%BE%9B%E5%BA%94%E5%95%86%E3%80%82%E5%AF%B9%E4%BA%8E%E9%82%A3%E4%BA%9B%E5%B8%8C%E6%9C%9B%E5%9C%A8%E5%8F%82%E8%B5%9B%E4%BD%9C%E5%93%81%E4%B8%AD%E5%8A%A0%E5%85%A5%E8%9B%8B%E7%99%BD%E8%B4%A8%E7%BB%93%E6%9E%84%E4%BF%A1%E6%81%AF%E7%9A%84%E5%8F%82%E8%B5%9B%E8%80%85%EF%BC%8C%E6%B0%A8%E5%9F%BA%E9%85%B8%E5%BA%8F%E5%88%97%E4%BD%8D%E4%BA%8EUniProt%E6%9D%A1%E7%9B%AE%E7%9A%842-555%E4%BD%8D[P34913](https://www.uniprot.org/uniprotkb/P34913/entry#sequences)%EF%BC%8C%E6%99%B6%E4%BD%93%E7%BB%93%E6%9E%84%E5%8F%AF%E4%BB%A5%E5%9C%A8[PDB%E6%9D%A1%E7%9B%AE3i28]%E4%B8%AD%E6%89%BE%E5%88%B0(https://www.rcsb.org/structure/3i28)%EF%BC%8C%E9%A2%84%E6%B5%8B%E7%BB%93%E6%9E%84%E5%8F%AF%E5%9C%A8AlphaFold2%E6%9D%A1%E7%9B%AE[34913]%E4%B8%AD%E6%89%BE%E5%88%B0(https://alphafold.ebi.ac.uk/entry/P34913">购自开曼化学</a>. 在PDB中可以发现具有结合配体的额外的EPHX2&#x2F;sEH晶体结构。</p>
<h3 id="BRD4"><a href="#BRD4" class="headerlink" title="BRD4"></a><strong>BRD4</strong></h3><p>The second target, bromodomain 4, is encoded by the BRD4 locus and its protein product is also named BRD4. Bromodomains bind to protein spools in the nucleus that DNA wraps around (called histones) and affect the likelihood that the DNA nearby is going to be transcribed, producing new gene products. Bromodomains play roles in cancer progression and a number of drugs <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10096006/">have been discovered to inhibit their activities</a>.</p>
<p>BRD4 <a target="_blank" rel="noopener" href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.9b01670">has been screened with DEL approaches previously</a> but the screening data were not published. We included BRD4 to allow contestants to evaluate candidate molecules for oncology indications.</p>
<p>We screened BRD4 <a target="_blank" rel="noopener" href="https://www.activemotif.com/catalog/details/31594/recombinant-brd4-44-460-protein">purchased from Active Motif</a>, a life sciences commercial vendor. For those contestants wishing to incorporate protein structural information in their submissions, the amino acid sequence is positions 44-460 from UniProt entry <a target="_blank" rel="noopener" href="https://www.uniprot.org/uniprotkb/O60885/entry#sequences">O60885-1</a>, the crystal structure (for a single domain) can be found in PDB entry <a target="_blank" rel="noopener" href="https://www.rcsb.org/structure/7USK">7USK</a> and predicted structure can be found in AlphaFold2 entry <a target="_blank" rel="noopener" href="https://alphafold.ebi.ac.uk/entry/O60885">O60885</a>. Additional BRD4 crystal structures with ligands bound can be found in PDB.</p>
<h3 id="ALB-HSA"><a href="#ALB-HSA" class="headerlink" title="ALB (HSA)"></a><strong>ALB (HSA)</strong></h3><p>The third target, serum albumin, is encoded by the ALB locus and its protein product is also named ALB. The protein product is sometimes abbreviated as HSA, for “human serum albumin”. ALB, the most <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/books/NBK459198/">common protein in the blood</a>, is used to drive osmotic pressure (to bring fluid back from tissues into blood vessels) and to transport many ligands, hormones, fatty acids, and more.</p>
<p>Albumin, being the most abundant protein in the blood, often plays a role in absorbing candidate drugs in the body and sequestering them from their target tissues. Adjusting candidate drugs to bind less to albumin and other blood proteins <a target="_blank" rel="noopener" href="https://pubs.acs.org/doi/10.1021/acsptsci.2c00115">is a strategy to help these candidate drugs be more effective</a>.</p>
<p>ALB <a target="_blank" rel="noopener" href="https://pubmed.ncbi.nlm.nih.gov/25650139/">has been screened with DEL approaches previously</a> but the screening data were not published. We included ALB to allow contestants to build models that might have a larger impact on drug discovery across many disease types. The ability to predict ALB binding well would allow drug developers to improve their candidate small molecule therapies much more quickly than physically manufacturing many variants and testing them against ALB empirically in an iterative process.</p>
<p>We screened ALB <a target="_blank" rel="noopener" href="https://www.abcam.com/products/proteins-peptides/recombinant-human-serum-albumin-protein-his-tag-ab217817.html">purchased from Active Motif</a>. For those contestants wishing to incorporate protein structural information in their submissions, the amino acid sequence is positions 25 to 609 from UniProt entry <a target="_blank" rel="noopener" href="https://www.uniprot.org/uniprotkb/P02768/entry#sequences">P02768</a>, the crystal structure can be found in PDB entry <a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/pdbe/entry/pdb/1AO6">1AO6</a>, and predicted structure can be found in AlphaFold2 entry <a target="_blank" rel="noopener" href="https://alphafold.ebi.ac.uk/entry/P02768">P02768</a>. Additional ALB crystal structures with ligands bound can be found in PDB.</p>
<p>Good luck!</p>
</div><div class="article-licensing box"><div class="licensing-title"><p>BELKA_2024</p><p><a href="https://hello-nilera.com/2024/05/19/BELKA-2024/">https://hello-nilera.com/2024/05/19/BELKA-2024/</a></p></div><div class="licensing-meta level is-mobile"><div class="level-left"><div class="level-item is-narrow"><div><h6>作者</h6><p>NilEra</p></div></div><div class="level-item is-narrow"><div><h6>发布于</h6><p>2024-05-19</p></div></div><div class="level-item is-narrow"><div><h6>更新于</h6><p>2024-07-29</p></div></div><div class="level-item is-narrow"><div><h6>许可协议</h6><p><a class="icons" rel="noopener" target="_blank" title="Creative Commons" href="https://creativecommons.org/"><i class="icon fab fa-creative-commons"></i></a><a class="icons" rel="noopener" target="_blank" title="Attribution" href="https://creativecommons.org/licenses/by/4.0/"><i class="icon fab fa-creative-commons-by"></i></a><a class="icons" rel="noopener" target="_blank" title="Noncommercial" href="https://creativecommons.org/licenses/by-nc/4.0/"><i class="icon fab fa-creative-commons-nc"></i></a></p></div></div></div></div></div><div class="article-tags is-size-7 mb-4"><span class="mr-2">#</span><a class="link-muted mr-2" rel="tag" href="/tags/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/">深度学习</a><a class="link-muted mr-2" rel="tag" href="/tags/%E6%9C%BA%E5%99%A8%E5%AD%A6%E4%B9%A0/">机器学习</a><a class="link-muted mr-2" rel="tag" href="/tags/PyTorch/">PyTorch</a><a class="link-muted mr-2" rel="tag" href="/tags/Kaggle/">Kaggle</a></div><div class="a2a_kit a2a_kit_size_32 a2a_default_style"><a class="a2a_dd" target="_blank" rel="noopener" href="https://www.addtoany.com/share"></a><a class="a2a_button_facebook"></a><a class="a2a_button_mastodon"></a><a class="a2a_button_email"></a></div><script async src="https://static.addtoany.com/menu/page.js"></script><script src="https://static.addtoany.com/menu/page.js" defer></script></article></div><div class="card"><div class="card-content"><h3 class="menu-label has-text-centered">喜欢这篇文章？打赏一下作者吧</h3><div class="buttons is-centered"><a class="button donate" href="https://afdian.net/a/nilera0001" target="_blank" rel="noopener" data-type="afdian"><span class="icon is-small"><i class="fas fa-charging-station"></i></span><span>爱发电</span></a><a class="button donate" data-type="alipay"><span class="icon is-small"><i class="fab fa-alipay"></i></span><span>支付宝</span><span class="qrcode"><img src="/" alt="支付宝"></span></a><a class="button donate" href="/" target="_blank" rel="noopener" data-type="buymeacoffee"><span class="icon is-small"><i class="fas fa-coffee"></i></span><span>送我杯咖啡</span></a><a class="button donate" href="/" target="_blank" rel="noopener" data-type="patreon"><span class="icon is-small"><i class="fab fa-patreon"></i></span><span>Patreon</span></a><a class="button donate" data-type="wechat"><span class="icon is-small"><i class="fab fa-weixin"></i></span><span>微信</span><span class="qrcode"><img src="/" alt="微信"></span></a></div></div></div><nav class="post-navigation mt-4 level is-mobile"><div class="level-start"><a class="article-nav-prev level level-item link-muted" href="/2024/05/22/Build-Tomcat/"><i class="level-item fas fa-chevron-left"></i><span class="level-item">IDEA 2022 搭建 Tomcat 环境</span></a></div><div class="level-end"><a class="article-nav-next level level-item link-muted" href="/2024/05/07/Tree/"><span class="level-item">Tree</span><i class="level-item fas fa-chevron-right"></i></a></div></nav><div class="card"><div class="card-content"><h3 class="title is-5">评论</h3><div id="comment-container"></div><link rel="stylesheet" href="https://cdnjs.loli.net/ajax/libs/gitalk/1.7.2/gitalk.css"><script src="https://cdnjs.loli.net/ajax/libs/gitalk/1.7.2/gitalk.min.js"></script><script>var gitalk = new Gitalk({
            id: "d7ebb48258491f4aaf6c106783007fba",
            repo: "NilEra-K.github.io",
            owner: "NilEra-K",
            clientID: "19819ad8edeef8a36f71",
            clientSecret: "75aca72740a16cc26a353794802276e27925fca7",
            admin: ["NilEra-K"],
            createIssueManually: false,
            distractionFreeMode: false,
            perPage: 20,
            pagerDirection: "last",
            
            
            enableHotKey: true,
            language: "zh-CN",
        })
        gitalk.render('comment-container')</script></div></div></div><div class="column column-left is-4-tablet is-4-desktop is-3-widescreen  order-1"><div class="card widget" data-type="profile"><div class="card-content"><nav class="level"><div class="level-item has-text-centered flex-shrink-1"><div><figure class="image is-128x128 mx-auto mb-2"><img class="avatar is-rounded" src="/img/avatarMyself.jpg" alt="NilEra"></figure><p class="title is-size-4 is-block" style="line-height:inherit;">NilEra</p><p class="is-size-6 is-block">C/C++ Developer!</p><p class="is-size-6 is-flex justify-content-center"><i class="fas fa-map-marker-alt mr-1"></i><span>Jinan Shandong</span></p></div></div></nav><nav class="level is-mobile"><div class="level-item has-text-centered is-marginless"><div><p class="heading">文章</p><a href="/archives"><p class="title">39</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">分类</p><a href="/categories"><p class="title">33</p></a></div></div><div class="level-item has-text-centered is-marginless"><div><p class="heading">标签</p><a href="/tags"><p class="title">38</p></a></div></div></nav><div class="level"><a class="level-item button is-primary is-rounded" href="https://github.com/NilEra-K" target="_blank" rel="noopener">关注我</a></div><div class="level is-mobile is-multiline"><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Github" href="https://github.com/NilEra-K"><i class="fab fa-github"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Facebook" href="https://facebook.com"><i class="fab fa-facebook"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Twitter" href="https://twitter.com"><i class="fab fa-twitter"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="Dribbble" href="https://dribbble.com"><i class="fab fa-dribbble"></i></a><a class="level-item button is-transparent is-marginless" target="_blank" rel="noopener" title="RSS" href="/"><i class="fas fa-rss"></i></a></div></div></div><div class="card widget" id="toc" data-type="toc"><div class="card-content"><div class="menu"><h3 class="menu-label">目录</h3><ul class="menu-list"><li><a class="level is-mobile" href="#Leash-Bio-Predict-New-Medicines-with-BELKA"><span class="level-left"><span class="level-item">1</span><span class="level-item">Leash Bio - Predict New Medicines with BELKA</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Overview"><span class="level-left"><span class="level-item">1.1</span><span class="level-item">Overview</span></span></a></li><li><a class="level is-mobile" href="#Description"><span class="level-left"><span class="level-item">1.2</span><span class="level-item">Description</span></span></a></li><li><a class="level is-mobile" href="#Evaluation"><span class="level-left"><span class="level-item">1.3</span><span class="level-item">Evaluation</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Submission-File"><span class="level-left"><span class="level-item">1.3.1</span><span class="level-item">Submission File</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Timeline"><span class="level-left"><span class="level-item">1.4</span><span class="level-item">Timeline</span></span></a></li><li><a class="level is-mobile" href="#Prizes"><span class="level-left"><span class="level-item">1.5</span><span class="level-item">Prizes</span></span></a></li><li><a class="level is-mobile" href="#Competition-Host"><span class="level-left"><span class="level-item">1.6</span><span class="level-item">Competition Host</span></span></a></li><li><a class="level is-mobile" href="#Additional-Details"><span class="level-left"><span class="level-item">1.7</span><span class="level-item">Additional Details</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#Chemical-Representations"><span class="level-left"><span class="level-item">1.7.1</span><span class="level-item">Chemical Representations</span></span></a></li><li><a class="level is-mobile" href="#SMILES"><span class="level-left"><span class="level-item">1.7.2</span><span class="level-item">SMILES</span></span></a></li><li><a class="level is-mobile" href="#Details-about-the-experiments"><span class="level-left"><span class="level-item">1.7.3</span><span class="level-item">Details about the experiments</span></span></a></li><li><a class="level is-mobile" href="#DELs-are-libraries-of-small-molecules-with-unique-DNA-barcodes-covalently-attached"><span class="level-left"><span class="level-item">1.7.4</span><span class="level-item">DELs are libraries of small molecules with unique DNA barcodes covalently attached</span></span></a></li><li><a class="level is-mobile" href="#DELs-are-manufactured-by-combining-different-building-blocks"><span class="level-left"><span class="level-item">1.7.5</span><span class="level-item">DELs are manufactured by combining different building blocks</span></span></a></li></ul></li><li><a class="level is-mobile" href="#Acknowledgements"><span class="level-left"><span class="level-item">1.8</span><span class="level-item">Acknowledgements</span></span></a></li><li><a class="level is-mobile" href="#Citation"><span class="level-left"><span class="level-item">1.9</span><span class="level-item">Citation</span></span></a></li><li><a class="level is-mobile" href="#Dataset-Description"><span class="level-left"><span class="level-item">1.10</span><span class="level-item">Dataset Description</span></span></a></li><li><a class="level is-mobile" href="#Overview-1"><span class="level-left"><span class="level-item">1.11</span><span class="level-item">Overview</span></span></a></li><li><a class="level is-mobile" href="#Files"><span class="level-left"><span class="level-item">1.12</span><span class="level-item">Files</span></span></a></li><li><a class="level is-mobile" href="#Competition-data"><span class="level-left"><span class="level-item">1.13</span><span class="level-item">Competition data</span></span></a></li><li><a class="level is-mobile" href="#Targets"><span class="level-left"><span class="level-item">1.14</span><span class="level-item">Targets</span></span></a><ul class="menu-list"><li><a class="level is-mobile" href="#EPHX2-sEH"><span class="level-left"><span class="level-item">1.14.1</span><span class="level-item">EPHX2 (sEH)</span></span></a></li><li><a class="level is-mobile" href="#BRD4"><span class="level-left"><span class="level-item">1.14.2</span><span class="level-item">BRD4</span></span></a></li><li><a class="level is-mobile" href="#ALB-HSA"><span class="level-left"><span class="level-item">1.14.3</span><span class="level-item">ALB (HSA)</span></span></a></li></ul></li></ul></li></ul></div></div><style>#toc .menu-list > li > a.is-active + .menu-list { display: block; }#toc .menu-list > li > a + .menu-list { display: none; }</style><script src="/js/toc.js" defer></script></div><div class="card widget" data-type="links"><div class="card-content"><div class="menu"><h3 class="menu-label">链接</h3><ul class="menu-list"><li><a class="level is-mobile" href="https://github.com/NilEra-K" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">GitHub</span></span><span class="level-right"><span class="level-item tag">github.com</span></span></a></li><li><a class="level is-mobile" href="https://nano.chemtian.top/" target="_blank" rel="noopener"><span class="level-left"><span class="level-item">Thymol Blue</span></span><span class="level-right"><span class="level-item tag">nano.chemtian.top</span></span></a></li></ul></div></div></div><div class="card widget" data-type="categories"><div class="card-content"><div class="menu"><h3 class="menu-label">分类</h3><ul class="menu-list"><li><a class="level is-mobile" href="/categories/CppDev/"><span class="level-start"><span class="level-item">CppDev</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/CppDev/Gaming/"><span class="level-start"><span class="level-item">Gaming</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/CppDev/QT6/"><span class="level-start"><span class="level-item">QT6</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/CppDev/%E5%A4%9A%E7%BA%BF%E7%A8%8B%E7%BC%96%E7%A8%8B/"><span class="level-start"><span class="level-item">多线程编程</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/CppDev/%E7%BD%91%E7%BB%9C%E7%BC%96%E7%A8%8B/"><span class="level-start"><span class="level-item">网络编程</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/CppDev/%E8%AF%AD%E6%B3%95%E7%82%B9/"><span class="level-start"><span class="level-item">语法点</span></span><span class="level-end"><span class="level-item tag">3</span></span></a><ul><li><a class="level is-mobile" href="/categories/CppDev/%E8%AF%AD%E6%B3%95%E7%82%B9/bind/"><span class="level-start"><span class="level-item">bind</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/CppDev/%E8%AF%AD%E6%B3%95%E7%82%B9/enum-class/"><span class="level-start"><span class="level-item">enum class</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/CppDev/%E8%AF%AD%E6%B3%95%E7%82%B9/%E8%99%9A%E5%87%BD%E6%95%B0/"><span class="level-start"><span class="level-item">虚函数</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></li><li><a class="level is-mobile" href="/categories/Linux/"><span class="level-start"><span class="level-item">Linux</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/Linux/%E8%BD%AF%E4%BB%B6%E6%BA%90/"><span class="level-start"><span class="level-item">软件源</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/%E5%A4%A7%E6%95%B0%E6%8D%AE%E6%8A%80%E6%9C%AF/"><span class="level-start"><span class="level-item">大数据技术</span></span><span class="level-end"><span class="level-item tag">8</span></span></a><ul><li><a class="level is-mobile" href="/categories/%E5%A4%A7%E6%95%B0%E6%8D%AE%E6%8A%80%E6%9C%AF/ECharts/"><span class="level-start"><span class="level-item">ECharts</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/%E5%A4%A7%E6%95%B0%E6%8D%AE%E6%8A%80%E6%9C%AF/HBase/"><span class="level-start"><span class="level-item">HBase</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/categories/%E5%A4%A7%E6%95%B0%E6%8D%AE%E6%8A%80%E6%9C%AF/Scala/"><span class="level-start"><span class="level-item">Scala</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/%E5%A4%A7%E6%95%B0%E6%8D%AE%E6%8A%80%E6%9C%AF/Spark/"><span class="level-start"><span class="level-item">Spark</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/%E5%A4%A7%E6%95%B0%E6%8D%AE%E6%8A%80%E6%9C%AF/%E6%95%B0%E6%8D%AE%E5%8F%AF%E8%A7%86%E5%8C%96/"><span class="level-start"><span class="level-item">数据可视化</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/%E5%A4%A7%E6%95%B0%E6%8D%AE%E6%8A%80%E6%9C%AF/%E9%A1%B9%E7%9B%AE/"><span class="level-start"><span class="level-item">项目</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/%E5%BC%80%E5%8F%91%E5%B7%A5%E5%85%B7/"><span class="level-start"><span class="level-item">开发工具</span></span><span class="level-end"><span class="level-item tag">4</span></span></a><ul><li><a class="level is-mobile" href="/categories/%E5%BC%80%E5%8F%91%E5%B7%A5%E5%85%B7/Axure-RP-9/"><span class="level-start"><span class="level-item">Axure RP 9</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/%E5%BC%80%E5%8F%91%E5%B7%A5%E5%85%B7/Docker/"><span class="level-start"><span class="level-item">Docker</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/%E5%BC%80%E5%8F%91%E5%B7%A5%E5%85%B7/Git/"><span class="level-start"><span class="level-item">Git</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/%E5%BC%80%E5%8F%91%E5%B7%A5%E5%85%B7/VSCode/"><span class="level-start"><span class="level-item">VSCode</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84-%E7%AE%97%E6%B3%95/"><span class="level-start"><span class="level-item">数据结构/算法</span></span><span class="level-end"><span class="level-item tag">4</span></span></a><ul><li><a class="level is-mobile" href="/categories/%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84-%E7%AE%97%E6%B3%95/%E5%93%88%E5%B8%8C-Hash/"><span class="level-start"><span class="level-item">哈希(Hash)</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84-%E7%AE%97%E6%B3%95/%E6%A0%88-Stack/"><span class="level-start"><span class="level-item">栈(Stack)</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li><li><a class="level is-mobile" href="/categories/%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84-%E7%AE%97%E6%B3%95/%E6%A0%91-Tree/"><span class="level-start"><span class="level-item">树(Tree)</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/"><span class="level-start"><span class="level-item">深度学习</span></span><span class="level-end"><span class="level-item tag">7</span></span></a><ul><li><a class="level is-mobile" href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/Kaggle/"><span class="level-start"><span class="level-item">Kaggle</span></span><span class="level-end"><span class="level-item tag">2</span></span></a></li><li><a class="level is-mobile" href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/LLMs/"><span class="level-start"><span class="level-item">LLMs</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li><li><a class="level is-mobile" href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/PyTorch/"><span class="level-start"><span class="level-item">PyTorch</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li><li><a class="level is-mobile" href="/categories/%E9%97%B2%E8%81%8A/"><span class="level-start"><span class="level-item">闲聊</span></span><span class="level-end"><span class="level-item tag">1</span></span></a><ul><li><a class="level is-mobile" href="/categories/%E9%97%B2%E8%81%8A/%E8%AE%A1%E5%88%92/"><span class="level-start"><span class="level-item">计划</span></span><span class="level-end"><span class="level-item tag">1</span></span></a></li></ul></li></ul></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">订阅更新</h3><form action="https://feedburner.google.com/fb/a/mailverify" method="post" target="popupwindow" onsubmit="window.open(&#039;https://feedburner.google.com/fb/a/mailverify?uri=&#039;,&#039;popupwindow&#039;,&#039;scrollbars=yes,width=550,height=520&#039;);return true"><input type="hidden" value="" name="uri"><input type="hidden" name="loc" value="en_US"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="订阅"></div></div></form></div></div></div><div class="card widget" data-type="subscribe-email"><div class="card-content"><div class="menu"><h3 class="menu-label">follow.it</h3><form action="" method="post" target="_blank"><div class="field has-addons"><div class="control has-icons-left is-expanded"><input class="input" name="email" type="email" placeholder="Email"><span class="icon is-small is-left"><i class="fas fa-envelope"></i></span></div><div class="control"><input class="button" type="submit" value="订阅"></div></div></form></div></div></div><div class="column-right-shadow is-hidden-widescreen"></div></div><div class="column column-right is-4-tablet is-4-desktop is-3-widescreen is-hidden-touch is-hidden-desktop-only order-3"><div class="card widget" data-type="recent-posts"><div class="card-content"><h3 class="menu-label">最新文章</h3><article class="media"><div class="media-content"><p class="date"><time dateTime="2024-08-18T02:35:39.000Z">2024-08-18</time></p><p class="title"><a href="/2024/08/18/Create-AN-Environment-of-Kaggle/">Create_AN_Environment_of_Kaggle</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2024-08-17T01:24:42.000Z">2024-08-17</time></p><p class="title"><a href="/2024/08/17/RSNA-2024/">RSNA-2024</a></p><p class="categories"><a href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/">深度学习</a> / <a href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/Kaggle/">Kaggle</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2024-08-15T09:21:54.000Z">2024-08-15</time></p><p class="title"><a href="/2024/08/15/Virtual-Destructor-IN-Cpp/">Virtual_Destructor_IN_Cpp</a></p><p class="categories"><a href="/categories/CppDev/">CppDev</a> / <a href="/categories/CppDev/%E8%AF%AD%E6%B3%95%E7%82%B9/">语法点</a> / <a href="/categories/CppDev/%E8%AF%AD%E6%B3%95%E7%82%B9/%E8%99%9A%E5%87%BD%E6%95%B0/">虚函数</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2024-08-13T12:19:01.000Z">2024-08-13</time></p><p class="title"><a href="/2024/08/13/Deploy-LLMs-ON-Linux/">Deploy_LLMs_ON_Linux</a></p><p class="categories"><a href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/">深度学习</a> / <a href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/LLMs/">LLMs</a></p></div></article><article class="media"><div class="media-content"><p class="date"><time dateTime="2024-08-12T15:10:23.000Z">2024-08-12</time></p><p class="title"><a href="/2024/08/12/Windows-Build-llama-cpp/">Windows Build llama.cpp</a></p><p class="categories"><a href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/">深度学习</a> / <a href="/categories/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/LLMs/">LLMs</a></p></div></article></div></div><div class="card widget" data-type="archives"><div class="card-content"><div class="menu"><h3 class="menu-label">归档</h3><ul class="menu-list"><li><a class="level is-mobile" href="/archives/2024/08/"><span class="level-start"><span class="level-item">八月 2024</span></span><span class="level-end"><span class="level-item tag">9</span></span></a></li><li><a class="level is-mobile" href="/archives/2024/07/"><span class="level-start"><span class="level-item">七月 2024</span></span><span class="level-end"><span class="level-item tag">6</span></span></a></li><li><a class="level is-mobile" href="/archives/2024/06/"><span class="level-start"><span class="level-item">六月 2024</span></span><span class="level-end"><span class="level-item tag">7</span></span></a></li><li><a class="level is-mobile" href="/archives/2024/05/"><span class="level-start"><span class="level-item">五月 2024</span></span><span class="level-end"><span class="level-item tag">10</span></span></a></li><li><a class="level is-mobile" href="/archives/2024/04/"><span class="level-start"><span class="level-item">四月 2024</span></span><span class="level-end"><span class="level-item tag">3</span></span></a></li><li><a class="level is-mobile" href="/archives/2024/03/"><span class="level-start"><span class="level-item">三月 2024</span></span><span class="level-end"><span class="level-item tag">4</span></span></a></li></ul></div></div></div><div class="card widget" data-type="tags"><div class="card-content"><div class="menu"><h3 class="menu-label">标签</h3><div class="field is-grouped is-grouped-multiline"><div class="control"><a class="tags has-addons" href="/tags/Axure-RP-9/"><span class="tag">Axure RP 9</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/C-11-%E6%96%B0%E6%A0%87%E5%87%86/"><span class="tag">C++ 11 新标准</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/CppDev/"><span class="tag">CppDev</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Docker/"><span class="tag">Docker</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ECharts/"><span class="tag">ECharts</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/EasyX/"><span class="tag">EasyX</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/GameDev/"><span class="tag">GameDev</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Git/"><span class="tag">Git</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/HBase/"><span class="tag">HBase</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Hadoop/"><span class="tag">Hadoop</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/JavaDev/"><span class="tag">JavaDev</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Kaggle/"><span class="tag">Kaggle</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/LLMs/"><span class="tag">LLMs</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/LM-Studio/"><span class="tag">LM-Studio</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Linux/"><span class="tag">Linux</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/MyBatis-Plus/"><span class="tag">MyBatis-Plus</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PowerInfer/"><span class="tag">PowerInfer</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/PyTorch/"><span class="tag">PyTorch</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/QT6/"><span class="tag">QT6</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/R/"><span class="tag">R</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Scala/"><span class="tag">Scala</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Spark/"><span class="tag">Spark</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/Tomcat/"><span class="tag">Tomcat</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/VSCode/"><span class="tag">VSCode</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/ZooKeeper/"><span class="tag">ZooKeeper</span><span class="tag">2</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E5%A4%9A%E7%BA%BF%E7%A8%8B%E7%BC%96%E7%A8%8B/"><span class="tag">多线程编程</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E5%A4%A7%E6%95%B0%E6%8D%AE%E6%8A%80%E6%9C%AF/"><span class="tag">大数据技术</span><span class="tag">8</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E5%BC%80%E5%8F%91%E5%B7%A5%E5%85%B7/"><span class="tag">开发工具</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84/"><span class="tag">数据结构</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E6%9C%BA%E5%99%A8%E5%AD%A6%E4%B9%A0/"><span class="tag">机器学习</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0/"><span class="tag">深度学习</span><span class="tag">7</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E7%AE%97%E6%B3%95/"><span class="tag">算法</span><span class="tag">4</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E7%BD%91%E7%BB%9C%E7%BC%96%E7%A8%8B/"><span class="tag">网络编程</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E8%AE%A1%E5%88%92/"><span class="tag">计划</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E8%AF%AD%E6%B3%95%E7%82%B9/"><span class="tag">语法点</span><span class="tag">3</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E8%BD%AF%E4%BB%B6%E6%BA%90/"><span class="tag">软件源</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E9%97%B2%E8%81%8A/"><span class="tag">闲聊</span><span class="tag">1</span></a></div><div class="control"><a class="tags has-addons" href="/tags/%E9%A1%B9%E7%9B%AE/"><span class="tag">项目</span><span class="tag">1</span></a></div></div></div></div></div></div></div></div></section><footer class="footer"><div class="container"><div class="level"><div class="level-start"><a class="footer-logo is-block mb-2" href="/"><img src="/img/StarLogo.svg" alt="Hello, NilEra :-)" height="28"></a><p class="is-size-7"><span>&copy; 2024 NilEra</span>  Powered by <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a> &amp; <a href="https://github.com/ppoffice/hexo-theme-icarus" target="_blank" rel="noopener">Icarus</a></p><p class="is-size-7">© 2024 前方⚡高能</p></div><div class="level-end"><div class="field has-addons"><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Creative Commons" href="https://creativecommons.org/"><i class="fab fa-creative-commons"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Attribution 4.0 International" href="https://creativecommons.org/licenses/by/4.0/"><i class="fab fa-creative-commons-by"></i></a></p><p class="control"><a class="button is-transparent is-large" target="_blank" rel="noopener" title="Download on GitHub" href="https://github.com/NilEra-K"><i class="fab fa-github"></i></a></p></div></div></div></div></footer><script src="https://cdnjs.loli.net/ajax/libs/jquery/3.3.1/jquery.min.js"></script><script src="https://cdnjs.loli.net/ajax/libs/moment.js/2.22.2/moment-with-locales.min.js"></script><script src="https://cdnjs.loli.net/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" defer></script><script>moment.locale("zh-cn");</script><script>var IcarusThemeSettings = {
            article: {
                highlight: {
                    clipboard: true,
                    fold: 'unfolded'
                }
            }
        };</script><script src="/js/column.js"></script><script src="/js/animation.js"></script><a id="back-to-top" title="回到顶端" href="javascript:;"><i class="fas fa-chevron-up"></i></a><script src="/js/back_to_top.js" defer></script><!--!--><!--!--><!--!--><script src="https://cdnjs.loli.net/ajax/libs/cookieconsent/3.1.1/cookieconsent.min.js" defer></script><script>window.addEventListener("load", () => {
      window.cookieconsent.initialise({
        type: "info",
        theme: "edgeless",
        static: false,
        position: "bottom-left",
        content: {
          message: "此网站使用Cookie来改善您的体验。",
          dismiss: "知道了！",
          allow: "允许使用Cookie",
          deny: "拒绝",
          link: "了解更多",
          policy: "Cookie政策",
          href: "https://www.cookiesandyou.com/",
        },
        palette: {
          popup: {
            background: "#edeff5",
            text: "#838391"
          },
          button: {
            background: "#4b81e8"
          },
        },
      });
    });</script><script src="https://cdnjs.loli.net/ajax/libs/lightgallery/1.10.0/js/lightgallery.min.js" defer></script><script src="https://cdnjs.loli.net/ajax/libs/justifiedGallery/3.8.1/js/jquery.justifiedGallery.min.js" defer></script><script>window.addEventListener("load", () => {
            if (typeof $.fn.lightGallery === 'function') {
                $('.article').lightGallery({ selector: '.gallery-item' });
            }
            if (typeof $.fn.justifiedGallery === 'function') {
                if ($('.justified-gallery > p > .gallery-item').length) {
                    $('.justified-gallery > p > .gallery-item').unwrap();
                }
                $('.justified-gallery').justifiedGallery();
            }
        });</script><!--!--><!--!--><script type="text/javascript" id="MathJax-script" async>MathJax = {
      tex: {
        inlineMath: [['$', '$'], ['\\(', '\\)']]
      },
      svg: {
        fontCache: 'global'
      },
      chtml: {
        matchFontHeight: false
      }
    };</script><script src="https://cdnjs.loli.net/ajax/libs/mathjax/3.2.2/es5/tex-mml-chtml.js"></script><!--!--><!--!--><!--!--><script src="/js/main.js" defer></script><div class="searchbox"><div class="searchbox-container"><div class="searchbox-header"><div class="searchbox-input-container" id="algolia-input"></div><div id="algolia-poweredby" style="display:flex;margin:0 .5em 0 1em;align-items:center;line-height:0"></div><a class="searchbox-close" href="javascript:;">×</a></div><div class="searchbox-body"></div><div class="searchbox-footer"></div></div></div><script src="https://cdnjs.loli.net/ajax/libs/algoliasearch/4.0.3/algoliasearch-lite.umd.js" crossorigin="anonymous" defer></script><script src="https://cdnjs.loli.net/ajax/libs/instantsearch.js/4.3.1/instantsearch.production.min.js" crossorigin="anonymous" defer></script><script src="/js/algolia.js" defer></script><script>document.addEventListener('DOMContentLoaded', function () {
            loadAlgolia({"applicationId":"2TB5ZZYPCO","apiKey":"00a43f1d62ca7b24c8b78d5f0223c065","indexName":"dev_nilera_blog"}, {"hint":"想要查找什么...","no_result":"未找到搜索结果","untitled":"(无标题)","empty_preview":"(无内容预览)"});
        });</script></body></html>